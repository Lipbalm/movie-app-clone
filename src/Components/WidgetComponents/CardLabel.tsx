import React, { FC } from "react";

interface IProps {
  genre_id: number;
  name: string;
}

const CardLabel: FC<IProps> = ({ genre_id, name }) => {
  const changeGenre = () => (genre_id === 13 ? "SF" : "unknown");
  return (
    <div>
      <div data-testid="genre">{changeGenre()}</div>
      <div data-testid="name">{name}</div>
    </div>
  );
};

export default CardLabel;
