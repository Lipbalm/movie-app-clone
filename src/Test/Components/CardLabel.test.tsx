import React from "react";
import { render } from "@testing-library/react";
import CardLabel from "../../Components/WidgetComponents/CardLabel";

const setup = () => {
  return { ...render(<CardLabel genre_id={13} name={"메트릭스"} />) };
};
describe("CardLabel", () => {
  it("SnapShot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("CardLabsel 장르 및 이름 보이기", () => {
    const { getByTestId } = setup();
    const genre = getByTestId("genre");
    const name = getByTestId("name");

    expect(genre).toHaveTextContent("SF");
    expect(name).toHaveTextContent("메트릭스");
  });
});
