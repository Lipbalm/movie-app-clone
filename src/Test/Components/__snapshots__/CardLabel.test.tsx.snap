// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLabel SnapShot 1`] = `
<div>
  <div>
    <div
      data-testid="genre"
    >
      SF
    </div>
    <div
      data-testid="name"
    >
      메트릭스
    </div>
  </div>
</div>
`;
